update-script.js

```diff
@@ -97,6 +100,75 @@ const checkIfNewRecords = (sourceRecords, destinationRecords) => {
   return areNewRecords;
 }
 
+/**
+ * Converts the records from the "password-rules" Remote Settings collection into a Map 
+ * for easier comparison against the GitHub source of truth records.
+ *
+ * @param {Object[]} records
+ * @param {string} records.Domain
+ * @param {string} records[password-rules]
+ * @return {Map} 
+ */
+const remoteSettingsRecordsToMap = (records) => {
+  let map = new Map();
+  for (let record of records) {
+    let { id, Domain: domain, "password-rules": rules } = record;
+    map.set(domain, { id: id, "password-rules": rules });
+  }
+  return map;
+}
+
+/**
+ * Creates and/or updates the existing records in the "password-rules" Remote Settings collection with the updated data from Apple's GitHub repository
+ *
+ * @param {KintoClient} client KintoClient instance
+ * @param {string} bucket Name of the Remote Settings bucket
+ */
+const createAndUpdateRulesRecords = async (client, bucket) => {
+  let githubRecords = await getSourceRecords(PASSWORD_RULES_API_ENDPOINT);
+  let { data: remoteSettingsRecords } = await client.bucket(bucket).collection(PASSWORD_RULES_COLLECTION_ID).listRecords();
+  let remoteSettingsMap = remoteSettingsRecordsToMap(remoteSettingsRecords);
+  let batchRecords = [];
+
+  for (let key in githubRecords) {
+    let comparisonRules = githubRecords[key]["password-rules"];
+    let oldRecord = remoteSettingsMap.get(key);
+    let oldRules = oldRecord?.["password-rules"];
+    if (!oldRecord) {
+      let newRecord = { "Domain": key, "password-rules": githubRecords[key]["password-rules"] };
+      batchRecords.push(newRecord);
+      console.log("Added new record to batch!", newRecord);
+    }
+    if (oldRecord && oldRules !== comparisonRules) {
+      let updatedRecord = { "id": oldRecord.id, "Domain": key, "password-rules": comparisonRules };
+      batchRecords.push(updatedRecord);
+      console.log("Added updated record to batch!", updatedRecord);
+    }
+
+  }
+  await client.bucket(bucket).collection(PASSWORD_RULES_COLLECTION_ID).batch(batch => {
```

Comment:
`client.bucket(bucket).collection(PASSWORD_RULES_COLLECTION_ID)`  could be stored as just `client`